packages/mask/src/social-network-adaptor/mirror.xyz/collecting/posts.ts

```diff
@@ -0,0 +1,162 @@
+import { DOMProxy, MutationObserverWatcher } from '@dimensiondev/holoflows-kit'
+import { creator, SocialNetworkUI as Next } from '../../../social-network/index.js'
+import { postsContentSelector } from '../utils/selectors.js'
+
+import { mirrorShared } from '../shared'
+import { createRefsForCreatePostContext } from '../../../social-network/utils/create-post-context'
+import { startWatch } from '../../../utils'
+import { formatWriter, getMirrorPageType, MirrorPageType } from './utils'
+import { Mirror } from '@masknet/web3-providers'
+import type { PostContextCoAuthor } from '@masknet/plugin-infra/content-script'
+
+const MIRROR_LINK_PREFIX = /https(.*)mirror.xyz(.*)\//i
+
+export function queryInjectPoint(node: HTMLElement) {
+    const allANode = node.querySelectorAll(
+        [
+            // if have header image is 4, either 3
+            ':scope > div:nth-child(3) div:has(div > div > div+a)~div',
+            ':scope > div:nth-child(4) div:has(div > div > div+a)~div',
+            ':scope > div:nth-child(3) div:has(div > div > div+a)~div~a',
+            ':scope > div:nth-child(4) div:has(div > div > div+a)~div~a',
+
+            // collection page card footer
+            ':scope > div:nth-child(1) footer span > div > div',
+            // if have header image is 4, either 3 entries: time + entry link
+            ':scope > div:nth-child(3) div+a',
+            ':scope > div:nth-child(4) div+a',
+
+            // if have header image is 4, either 3 entry: address link + time
+            ':scope > div:nth-child(3) > div > div > a~div',
+            ':scope > div:nth-child(4) > div > div > a~div',
+        ].join(),
+    )
+    return allANode.item(allANode.length - 1) as HTMLElement
+}
+
+const getPostId = (node: HTMLElement | HTMLLinkElement) => {
+    // Handle entry detail page post id
+    if (getMirrorPageType(location.href) === MirrorPageType.Post) {
+        return location.pathname.match(/\w{43}/i)?.[0]
+    }
+
+    const ele = node.querySelector<HTMLLinkElement>('div > a')
+    const href = ele?.href || (node as HTMLLinkElement)?.href
+
+    if (href?.startsWith('https')) {
+        return href.replace(MIRROR_LINK_PREFIX, '')
+    }
+
+    if (href) return href?.replace('/', '')
+
+    return ''
+}
+
+const getPostWriters = async (postId: string) => {
+    const script = document.getElementById('__NEXT_DATA__')?.innerHTML
+    if (!script) return
+    const INIT_DATA = JSON.parse(script)
+
+    function getAuthorDetail(address?: string) {
+        const author = INIT_DATA?.props?.pageProps?.__APOLLO_STATE__?.[`ProjectType:${address}`]
```

Comment:
This might be incorrect on client side rendering.

- open `page1` (SSR)
- client side navigate to `page2`
- fetching data for `page2`, the fetched data will not matching initial data embed in page.

Reply:
If DOM change, will reload data:
```typescript
return {
  onNodeMutation: run,
  onTargetChanged: run,
  onRemove: () => postStore.delete(proxy),
}
```

Reply:
OK. That will be fine